typescript/phoenix-sdk/src/utils/market.ts

```diff
@@ -360,10 +399,12 @@ function levelToUiLevel(
   quoteAtomsPerQuoteUnit: number
 ): [number, number] {
   return [
-    (toNum(priceInTicks) / quoteAtomsPerQuoteUnit) *
+    ((toNum(priceInTicks) / quoteAtomsPerQuoteUnit) *
```

Comment:
conversion fixed to return sizes in raw base units (not base units) and prices in quote units per raw base unit (not quote units per base unit), please double-check @jarry-xiao

Reply:
Units check out, I'm not sure about the rounding though. Would it be safer to do everything in BN and cast down after

Reply:
I don't think so, because these end up as non-integers. BN only supports integer division. This is for levels so it doesn't matter if two orders have slightly different values here, but may have some weirdness for the L3 UI conversion. I *think* those should be fine too since we round later and don't reorder.